sdk/python/feast/feature_store.py

```diff
@@ -1002,17 +991,78 @@ def get_online_features(
                 union_of_entity_keys,
             )
 
+        requested_result_row_names = self._get_requested_result_fields_and_populate_odfv_dependencies(
+            entity_name_to_join_key_map,
+            full_feature_names,
+            grouped_odfv_refs,
+            provider,
+            request_data_features,
+            needed_request_fv_features,
+            result_rows,
+            union_of_entity_keys,
+        )
+
         initial_response = OnlineResponse(
             GetOnlineFeaturesResponse(field_values=result_rows)
         )
         return self._augment_response_with_on_demand_transforms(
             _feature_refs,
-            all_on_demand_feature_views,
+            requested_result_row_names,
+            requested_on_demand_feature_views,
             full_feature_names,
             initial_response,
             result_rows,
         )
 
+    def _get_requested_result_fields_and_populate_odfv_dependencies(
```

Comment:
`and` in a method name tends to be a bit of a code smell to me. I might be misunderstanding but it looks like this could be split into two separate methods?

* `_get_requested_result_fields`
* `_populate_odfv_dependencies`

Reply:
/nit Separate comment as this one is my own personal gripe - I'm not a fan of side effects which seem to be the way you're updating the `result_rows` relying on the fact that a Python list is mutable. 'Tis fine but I typically find that it's harder to work out what's going on compared to populating and returning a new list. There could be an argument for this way though - memory maybe?

Reply:
Fixed the first bit. Yeah I'm also not a fan of populating as a side effect. 

Though in this case, its actually not relying on the python list being mutable and more relying on the underlying proto builder, so in this case does seem slightly canonical.